### DISCUSSION TITLE
[REQ] Wish 중복 저장 해결

### DISCUSSION THREAD

**Discussion by Dr-KoKo** [Strong-Potato/TripVote-BE#115] [General]
## 기능

Wish의 중복 저장을 방지한다.
 

## 설명

https://github.com/Strong-Potato/TripVote-BE/blob/2769d16ff2ee2c3c08a42fc3ff1c5f46b8b3a8f2/app/src/main/java/fc/be/app/domain/wish/service/WishService.java#L26

Wish를 저장할 떄 DB에 동일한 Wish가 있는지 확인하는 로직이 부재인 상태입니다. 이 때 두 번 이상의 wish/add 요청이 오게되면 wish에는 동일한 member, place 에 대해 두 개 이상의 wish가 저장되게 됩니다. 그렇게 되면 향후 getWishes 메서드를 호출할 때

https://github.com/Strong-Potato/TripVote-BE/blob/2769d16ff2ee2c3c08a42fc3ff1c5f46b8b3a8f2/app/src/main/java/fc/be/app/domain/wish/service/WishService.java#L40

중복된 Wish들이 섞여서 반환되게 됩니다. 따라서 add 할 때 중복을 방지하기 위해 현재 유저가 요청 받은 place에 찜한 이력이 있는지 확인하는 방어 로직을 추가하면 좋을 것 같습니다.

## 기대 효과

Wish가 중복 저장되는 오류를 방지할 수 있다.

**Comment by laigasus** [comment]
@Dr-KoKo  피드백 주셔서 감사합니다 🙇

autoincrement의 기본키를 없애고, 복합키를 이용하여 선언했기 때문에
한 회원에 대해 중복된 장소가 저장되진 않습니다.

이미 존재하는 값에 대해서는 update로 자동 수행(1회 쿼리 수행)합니다.
존재하는지 확인하고 추가하는 수행(2회 쿼리 수행) 보다 db 접근 횟수가 적다고 생각했기 때문입니다.

현 시점에서의 기능 결함이 있을까요?

https://github.com/Strong-Potato/TripVote-BE/blob/bebcc8de568e11679a2399d730cad2ce0d9d6a3b/app/src/main/java/fc/be/app/domain/wish/entity/WishId.java#L8-L17

https://github.com/Strong-Potato/TripVote-BE/blob/bebcc8de568e11679a2399d730cad2ce0d9d6a3b/app/src/main/java/fc/be/app/domain/wish/entity/Wish.java#L12-L28

![image](https://github.com/Strong-Potato/TripVote-BE/assets/26517061/7f7a7556-84f7-4043-9bae-9c96fa31e9c9)

마구 insert 실행해도 물리적 데이터는 그대로인 모습입니다.
![image](https://github.com/Strong-Potato/TripVote-BE/assets/26517061/df0935a0-64fa-4f9c-9678-68047f4fd1a7)

**Comment by Dr-KoKo** [comment]
네넵 확인했습니다. 복합키로 되어있는 부분을 확인 못했네요. 확인해주셔서 감사합니다.
